lib/filterx/filterx-expr.c

```diff
@@ -88,7 +88,9 @@ filterx_expr_optimize(FilterXExpr *self)
   optimized = filterx_expr_optimize(optimized);
 
   msg_trace("FilterX: expression optimized",
-            filterx_expr_format_location_tag(self));
+            filterx_expr_format_location_tag(self),
+            evt_tag_str("old_type", self->type),
+            evt_tag_str("new_type", optimized->type));
```

Comment:
I don't see `type` ever get a value, in fact commenting it out from the struct builds just fine. I think we should fill the `type` field for every `Expr` before using it in logs.

Reply:
I'm extracting that commit into a separate PR, as I have a number of subsequent commits that builds on top of that.

Reply:
I have now rebased this against #444, which means this contains that patch in addition to master. With that this should compile and once that patch goes in, the rest of my patches in #434 can move up to.